subprojects/plugins/src/main/java/org/gradle/api/plugins/TestReportAggregationPlugin.java

```diff
@@ -0,0 +1,121 @@
+
+    @Override
+    public void apply(Project project) {
+        project.getPluginManager().apply("org.gradle.reporting-base");
+
+        final Configuration testAggregation = project.getConfigurations().create(TEST_REPORT_AGGREGATION_CONFIGURATION_NAME);
+        testAggregation.setDescription("A configuration to collect test execution results");
+        testAggregation.setVisible(false);
+        testAggregation.setCanBeConsumed(false);
+        testAggregation.setCanBeResolved(false);
+
+        ReportingExtension reporting = project.getExtensions().getByType(ReportingExtension.class);
+        reporting.getReports().registerBinding(AggregateTestReport.class, DefaultAggregateTestReport.class);
+
+        ObjectFactory objects = project.getObjects();
+
+        // prepare testReportDir with a reasonable default, but override with JavaPluginExtension#testReportDir if available
+        final DirectoryProperty testReportDir = objects.directoryProperty().convention(reporting.getBaseDirectory().dir(TestingBasePlugin.TESTS_DIR_NAME));
+        JavaPluginExtension javaPluginExtension = project.getExtensions().findByType(JavaPluginExtension.class);
+        if (javaPluginExtension != null) {
+            testReportDir.set(javaPluginExtension.getTestReportDir());
+        }
+
+        // iterate and configure each user-specified report, creating a <reportName>ExecutionData configuration for each
+        reporting.getReports().withType(AggregateTestReport.class).configureEach(report -> {
+            report.getReportTask().configure(task -> {
+
+                // A resolvable configuration to collect test results; typically named "testResults"
+                Configuration testResultsConf = project.getConfigurations().create(report.getName() + "Results");
+                testResultsConf.extendsFrom(testAggregation);
+                testResultsConf.setDescription(String.format("Supplies test result data to the %s.  External library dependencies may appear as resolution failures, but this is expected behavior.", report.getName()));
+                testResultsConf.setVisible(false);
+                testResultsConf.setCanBeConsumed(false);
+                testResultsConf.setCanBeResolved(true);
+                testResultsConf.attributes(attributes -> {
+                    attributes.attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category.class, Category.DOCUMENTATION));
+                    attributes.attribute(DocsType.DOCS_TYPE_ATTRIBUTE, objects.named(DocsType.class, DocsType.TEST_RESULTS));
+                    attributes.attributeProvider(TestType.TEST_TYPE_ATTRIBUTE, report.getTestType().map(tt -> objects.named(TestType.class, tt)));
+                    attributes.attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage.class, Usage.VERIFICATION));
+                });
+
+                Callable<FileCollection> testResults = () ->
+                    testResultsConf.getIncoming().artifactView(view -> {
```

Comment:
This artifactView assemble definitely seems like it ought to be a new method on configuration with a name describing its purpose.

Reply:
Probably, but I'd rather not solidify that until we get some real-world feedback.

Also, wrapping this in a callable was only necessary since the attribute values cannot (yet) be evaluated lazily. So this is subject to change.